client/components/site-selector/index.jsx

```diff
@@ -295,6 +286,78 @@ class SiteSelector extends Component {
 			sites = sites.filter( site => site.slug !== this.props.selected );
 		}
 
+		return sites;
+	}
+
+	shouldRenderRecentSites() {
+		return (
+			this.props.showRecentSites &&
+			this.shouldShowGroups() &&
+			! this.props.sitesFound &&
+			size( this.props.recentSites )
```

Comment:
The check for non-empty `recentSites` can be removed. Even non-empty list can be made empty by a filter. `renderRecentSites` must do the check once more anyway.